packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -1,225 +1,211 @@
-      onClick();
-    }
-  };
+interface ChildProps {
+  className?: string;
+  disabled?: boolean;
+  children?: React.ReactNode;
+
+  /** @deprecated Use `className` to add new styles */
+  styles?: ReturnType<typeof getCardStyles>;
 }
 
-/**
- * @public
- */
-export const getCardStyles = stylesFactory((theme: GrafanaTheme2) => {
-  return {
-    inner: css`
-      display: flex;
-      justify-content: space-between;
-      align-items: center;
-      width: 100%;
-      flex-wrap: wrap;
-    `,
-    heading: css`
-      display: flex;
-      justify-content: space-between;
-      align-items: center;
-      width: 100%;
-      margin-bottom: 0;
-      font-size: ${theme.typography.size.md};
-      letter-spacing: inherit;
-      line-height: ${theme.typography.body.lineHeight};
-      color: ${theme.colors.text.primary};
-      font-weight: ${theme.typography.fontWeightMedium};
-    `,
-    info: css`
-      display: flex;
-      flex-direction: row;
-      justify-content: space-between;
-      align-items: center;
-      width: 100%;
-    `,
-    metadata: css`
-      display: flex;
-      align-items: center;
-      width: 100%;
-      font-size: ${theme.typography.size.sm};
-      color: ${theme.colors.text.secondary};
-      margin: ${theme.spacing(0.5, 0, 0)};
-      line-height: ${theme.typography.bodySmall.lineHeight};
-      overflow-wrap: anywhere;
-    `,
-    description: css`
-      width: 100%;
-      margin: ${theme.spacing(1, 0, 0)};
-      color: ${theme.colors.text.secondary};
-      line-height: ${theme.typography.body.lineHeight};
-    `,
-    media: css`
-      margin-right: ${theme.spacing(2)};
-      width: 40px;
+/** Main heading for the card */
+const Heading = ({ children, className, 'aria-label': ariaLabel }: ChildProps & { 'aria-label'?: string }) => {
+  const context = useContext(CardContext);
+  const styles = useStyles2(getHeadingStyles);
 
-      & > * {
-        width: 100%;
-      }
+  const { href, onClick } = context ?? { href: undefined, onClick: undefined };
+  const Wrapper = href ? 'a' : onClick ? 'button' : React.Fragment;
 
-      &:empty {
-        display: none;
-      }
-    `,
-    actionRow: css`
-      display: flex;
-      justify-content: space-between;
-      align-items: center;
-      width: 100%;
-      margin-top: ${theme.spacing(2)};
-    `,
-    actions: css`
-      & > * {
-        margin-right: ${theme.spacing(1)};
-      }
-    `,
-    secondaryActions: css`
-      display: flex;
-      align-items: center;
-      color: ${theme.colors.text.secondary};
-      // align to the right
-      margin-left: auto;
-      & > * {
-        margin-right: ${theme.spacing(1)} !important;
-      }
-    `,
-    separator: css`
-      margin: 0 ${theme.spacing(1)};
-    `,
-    tagList: css`
-      max-width: 50%;
-    `,
-  };
-});
+  return (
+    <h2 className={cx(styles.heading, className)}>
+      <Wrapper href={href} onClick={onClick} className={styles.linkHack} aria-label={ariaLabel}>
```

Comment:
I think this might need some conditional rendering as currently there's an error in console ```Warning: Invalid prop `onClick` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.``` I think even if href and/or onClick are undefined, they still trigger this error when passed to Fragment.